nengo_loihi/simulator.py

```diff
@@ -287,7 +295,8 @@ def __init__(  # noqa: C901
             precompute=False,
             target=None,
             progress_bar=None,
-            remove_passthrough=True
+            remove_passthrough=True,
+            hw_opts=None,
```

Comment:
Personally, I would get rid of `hw_opts` for now since it's not necessary for this, and if you think there's a case for it now make a separate issue/PR. It will be easier to get this PR through without having to discuss it.

Reply:
I think it should still be exposed somehow though. One of the tests compares the end-to-end results using a bunch of different allocators to make sure they're all the same. And the examples that we demonstrate next or any profiling that we do would depend on having this as some sort of switch.

Reply:
I agree. But it's been a contentious issue in the past, and is not part of the scope of this work. It needs to be broken off into a separate issue.

Reply:
Actually, this is required, since we currently do the build when the simulator is created, not when `run` is first called (which is when we set up snips). So I think this is a good case for having something like `hw_opts` (though I'm not sold on that name). What if we call it `hardware_args`? `hardware_opts` would be okay too, but I think `args` is more common than `opts` (though it does also have other connotations like `*args`).

Reply:
If we wanted to use [scipy.optimize](https://docs.scipy.org/doc/scipy/reference/tutorial/optimize.html) as a model example, then maybe `hardware_options`? Like you said `args` could be confused with an argument list `*args`. Another possibility would be to stick this into `nengo.rc`?

Reply:
My guess is that we'll have one allocator that people will use 99% of the time, and it will just be for special models that you'd need a different one. So I think having a simulator arg makes more sense. I'd be happy with `hardware_options`.

Reply:
Resolved by changing to `hardware_options`. We've added a backlog item for rethinking global simulator configuration within Nengo core.